dataproc_jupyter_plugin/services/bigqueryService.py

```diff
@@ -160,6 +155,56 @@ def bigquery_preview_data(self, credentials, dataset_id, table_id, page_token, l
             log.exception(f"Error fetching preview data")
             return {"error": str(e)}
 
+class BigQuerySearchService:
+    def bigquery_search(self, credentials, search_string, type, system, projects, log):
+        try:
+            if (
+                ("access_token" in credentials)
+                and ("project_id" in credentials)
+                and ("region_id" in credentials)
+            ):
+                access_token = credentials["access_token"]
+                project_id = credentials['project_id']
+                api_endpoint = (
+                    f"{datacatalog_url}v1/catalog:search"
+                )
+                headers = {
+                "Content-Type": CONTENT_TYPE,
+                "Authorization": f"Bearer {access_token}",
+                'X-Goog-User-Project' : project_id
+                }
```

Comment:
This formatting looks wrong. Did you remember to run the `black` command line tool over the modified files?